core/content/connector.js

```diff
@@ -285,137 +328,169 @@ var BaseConnector = window.BaseConnector || function () {
 				newTrack = artistTrack.track;
 			}
 
-			if (newTrack !== currentState.track) {
-				currentState.track = newTrack;
-				changedFields.push('track');
-			}
+			newAlbum = this.getAlbum();
 
-			if (newArtist !== currentState.artist) {
-				currentState.artist = newArtist;
-				changedFields.push('artist');
-			}
+			newDuration = this.getDuration();
 
-			var newAlbum = this.getAlbum();
-			if (newAlbum !== currentState.album) {
-				currentState.album = newAlbum;
-				changedFields.push('album');
-			}
+		} else {
+			console.info('This is a playlist.',this.currentPlaylist);
+			/**
+			 * This media is a collection of consecutive tracks.
+			*/
+			var newPlaylistTrack = this.playlistIncrementCheck();
 
-			var newUID = this.getUniqueID();
-			if (newUID !== currentState.uniqueID) {
-				currentState.uniqueID = newUID;
-				changedFields.push('uniqueID');
-			}
+			if(newPlaylistTrack) {
+				newTrack = newPlaylistTrack.track;
 
-			var newDuration = this.getDuration();
-			if (newDuration !== currentState.duration) {
-				currentState.duration = newDuration;
-				changedFields.push('duration');
-			}
+				newArtist = newPlaylistTrack.artist;
 
-			var newCurrentTime = this.getCurrentTime();
-			if (newCurrentTime !== currentState.currentTime) {
-				currentState.currentTime = newCurrentTime;
-				changedFields.push('currentTime');
-			}
+				newAlbum = this.getAlbum();
 
-			var newIsPlaying = this.isPlaying();
-			if (newIsPlaying !== currentState.isPlaying) {
-				currentState.isPlaying = newIsPlaying;
-				changedFields.push('isPlaying');
-			}
+				artistTrack = this.getArtistTrack();
+				if ( (newArtist === null || typeof newArtist === 'undefined') && artistTrack.artist) {
+					newArtist = artistTrack.artist;
+				}
+				if ( (newAlbum === null || typeof newArtist === 'undefined') && artistTrack.track) {
+					newAlbum = artistTrack.track; // Format will probably be Artist - Album Name, so hijack it.
+				}
 
-			var newTrackArt = this.getTrackArt();
-			if (newTrackArt !== currentState.trackArt) {
-				currentState.trackArt = newTrackArt;
-				changedFields.push('trackArt');
-			}
+				var newArtistTrack = newPlaylistTrack.artistTrack;
+				var separator = this.findSeparator(newArtistTrack);
+				if (separator !== null) {
+					if(newPlaylistTrack.artist) { newArtist = newArtistTrack.substr(0, separator.index); }
+					if(newPlaylistTrack.track)  { newTrack = newArtistTrack.substr(separator.index + separator.length); }
+				}
 
-			// take action if needed
-			if (changedFields.length > 0 && this.reactorCallback !== null) {
-				this.reactorCallback(currentState, changedFields);
+				var trackIndex = this.currentPlaylist.indexOf(newPlaylistTrack);
+				newDuration = newPlaylistTrack.startTime + (this.currentPlaylist[trackIndex+1].startTime || 20 );
```

Comment:
Give UID for playlist row, ontop of track UID.

`
if(newUID) {
   newUID += "#"+trackIndex
}
`